src/features/smSystem/suppliersOrders/components/AddOrderModal/Modal.tsx

```diff
@@ -0,0 +1,190 @@
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Modal,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { GridRowParams } from '@mui/x-data-grid';
+import { SetStateAction, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Pages } from '../../../../../utils';
+import {BasicModalProps} from '../../../app/types/index';
+import { useCreateOrder } from '../../api/useCreateOrder';
+import { useGetSuppliers } from '../../api/useGetSuppliers';
+
+import BranchesTable from './BranchesTable';
+import SuppliersTable from './SuppliersTable';
+
+
+const AddOrderModal = ({ open, handleClose }: BasicModalProps) => {
+  const [page, setPage] = useState(0);
+  const [name, setName] = useState<string>('');
+  const [nameToSearch, setNameToSearch] = useState<string>('');
+  const [isCreating, setIsCreating] = useState<boolean>(false);
+
+  const {
+    data,
+    isLoading,
+    isFetchingNextPage,
+    fetchNextPage,
+    isError,
+    name: searchedName,
+  } = useGetSuppliers(nameToSearch);
+  const [selectedSupplier, setSelectedSupplier] = useState<number | null>(null);
+  const [selectedBranches, setSelectedBranches] = useState<number[]>([]);
+  
+  const { createOrder } = useCreateOrder();
+
+  const navigate = useNavigate();
+
+  const handleAddOrder = async () => {
+    if (selectedSupplier !== null && selectedBranches.length > 0) {
+      setIsCreating(true)
+      const orderData = {
+        supplier_id: selectedSupplier,
+        selected_branches_ids: selectedBranches,
+      };
+  
+      try {
+        const id = await createOrder(orderData);
+        if (id) {
+          setIsCreating(false)
```

Comment:
`isCreating` state looks useless when you can extract this boolean from useCreateOrder hook